src/daemon/daemon.js

```diff
@@ -168,9 +168,53 @@ const parseCfgMultiaddr = (addr) => (addr.includes('/http')
   : multiaddr(addr).encapsulate('/http')
 )
 
+async function checkPortsArray (addrs) {
+  for (const addr of addrs) {
```

Comment:
Filter `null`/`''`? I think go-ipfs errors on startup if you do this but in js-ipfs you can completely disable the Gateway/API from starting in this way.

```suggestion
  addrs = addrs.filter(Boolean)
  for (const addr of addrs) {
```